src/rcore.c

```diff
@@ -3775,10 +3779,19 @@ bool IsKeyReleased(int key)
 // Check if a key is NOT being pressed (key not held down)
 bool IsKeyUp(int key)
 {
+    if ((key < 0) || (key >= sizeof(CORE.Input.Keyboard.currentKeyState))) return false;
     if (CORE.Input.Keyboard.currentKeyState[key] == 0) return true;
     else return false;
 }
 
+// Check if a key has been repeated
+bool IsKeyRepeated(int key)
+{
+    if ((key < 0) || (key >= sizeof(CORE.Input.Keyboard.currentKeyState))) return false;
```

Comment:
Not important, but some comments:
Should be `sizeof(repeatKeyState)`. But then, use another array every time? Checking with `MAX_KEYBOARD_KEYS` instead could make things consistent.

That said, I think these should be asserts instead. If you call this some invalid value this is a silent failure.
```
bool static IsValidKey(int key) {
return key >= 0 && key < MAX_KEYBOARD_KEYS);
}
// and using it
assert(isValidKey(key));
```

I have no objections with merging this one instead, since more platforms seem to work as well!

Reply:
> Checking with MAX_KEYBOARD_KEYS instead could make things consistent.

Agree.

 > I think these should be asserts instead. If you call this some invalid value this is a silent failure.

I'm afraid raylib does not use `assert()` anywhere. It can just `TraceLog(LOG_WARNING, ...)` if really required.